### PR TITLE
CT-884 migrate Workspaces to JSON

### PR DESCRIPTION
**Pull request by martinjunger**
Jira: CT-884

Before asking for review make sure that:

## Checklist

- [x] New client method(s) has tests
- [x] Apiary file is updated

## Release

  - [x] I gave the PR a proper label:
    * major (BC break)
    * minor (new feature)
    * patch (backwards compatible fix)
    * no release (just test changes)

### CODE REVIEW

**Review by jirkasemmler — CHANGES_REQUESTED** [review]
opet nerozumim stejnym vecem
- bojim se dopadu tech assertu
- proc nevracime nic tady ani z POST requestu?

File: src/Keboola/StorageApi/Workspaces.php [hunk 1]
```diff
@@ -61,27 +79,33 @@ public function deleteWorkspace($id, array $options = [], bool $async = false):
...
-     * @param array $options -- required input[mappings], optional preserve
-     * @return mixed|string
+     * @param int $id
+     * @param array $options input[mappings] (required), preserve (optional)
+     * @return void
      */
     public function cloneIntoWorkspace($id, array $options = [])
     {
-        return $this->client->apiPost("workspaces/{$id}/load-clone", $options);
+        $this->client->apiPostJson("workspaces/{$id}/load-clone", $options);
```
Comment (martinjunger) [inline src/Keboola/StorageApi/Workspaces.php:98]: > proc nevracime nic tady ani z POST requestu?

Protoze se realne nic nevrati:
- vytvori se job
- vrati se `202 Accepted`
- `Client::request` [pocka](https://github.com/keboola/storage-api-php-client/blob/47a34f1d425d361675f92b76643904d626db165c/src/Keboola/StorageApi/Client.php#L2515-L2518) az job dobehne
    ```
        // wait for asynchronous task completion
        if ($handleAsyncTask && $response->getStatusCode() == 202) {
            return $this->handleAsyncTask($response);
        }
     ```
- potom vrati vysledek
- ❗ ale tenhle [job nic nevraci](https://github.com/keboola/connection/blob/b38947b19c53a4ff3ec0b474a51c698672ec5211/legacy-app/application/modules/storage/jobs/WorkspaceLoad.php#L77-L77), zaroven se s vysledkem nikde v testech nepocita

File: src/Keboola/StorageApi/Workspaces.php [hunk 2]
```diff
@@ -61,27 +79,33 @@ public function deleteWorkspace($id, array $options = [], bool $async = false):
...
-     * @return mixed|string
+     * @param int $id
+     * @param array $options input[mappings] (required), preserve (optional)
+     * @return void
      */
     public function loadWorkspaceData($id, array $options = [])
     {
-        return $this->client->apiPost("workspaces/{$id}/load", $options);
+        $this->client->apiPostJson("workspaces/{$id}/load", $options);
     }
```
Comment (martinjunger) [inline src/Keboola/StorageApi/Workspaces.php:89]: > proc nevracime nic tady ani z POST requestu?

Protoze se realne nic nevrati - viz komentar pobliz...

**Comment by martinjunger** [comment]
ℹ️ Rebase

**Review by jirkasemmler — APPROVED** [review]
👍

**Comment by martinjunger** [comment]
[Tag](https://dev.azure.com/keboola-dev/Connection%20build/_build/results?buildId=25873&view=logs&j=ae13dfb3-00c3-5552-9b00-63e26024fcba&t=48e25ff6-be86-558e-50b2-b8f91a4d5f0f&l=1770) spadnul jen na:
```
15:13:14-UTC - paratest-backend-snowflake-abs-part-1  -	There were 4 errors:
15:13:14-UTC - paratest-backend-snowflake-abs-part-1  -	
15:13:14-UTC - paratest-backend-snowflake-abs-part-1  -	1) Keboola\Test\Backend\Workspaces\LegacyWorkspacesSnowflakeTest::testLoadIncrementalNotNullable
15:13:14-UTC - paratest-backend-snowflake-abs-part-1  -	Keboola\StorageApi\ClientException: Table in.c-API-tests-1eb2aae730941c1153220150a9cd2609f3bef5b7.languages load error: Likely datatype conversion: preg_match() expects parameter 2 to be string, int given
15:13:14-UTC - paratest-backend-snowflake-abs-part-1  -	
15:13:14-UTC - paratest-backend-snowflake-abs-part-1  -	/code/src/Keboola/StorageApi/Client.php:2563
15:13:14-UTC - paratest-backend-snowflake-abs-part-1  -	/code/src/Keboola/StorageApi/Client.php:2552
15:13:14-UTC - paratest-backend-snowflake-abs-part-1  -	/code/src/Keboola/StorageApi/Client.php:2520
15:13:14-UTC - paratest-backend-snowflake-abs-part-1  -	/code/src/Keboola/StorageApi/Client.php:2407
15:13:14-UTC - paratest-backend-snowflake-abs-part-1  -	/code/src/Keboola/StorageApi/Workspaces.php:88
15:13:14-UTC - paratest-backend-snowflake-abs-part-1  -	/code/tests/Backend/Workspaces/LegacyWorkspacesSnowflakeTest.php:114
15:13:14-UTC - paratest-backend-snowflake-abs-part-1  -	/code/vendor/keboola/phpunit-retry-annotations/src/RetryTrait.php:32
15:13:14-UTC - paratest-backend-snowflake-abs-part-1  -	
15:13:14-UTC - paratest-backend-snowflake-abs-part-1  -	2) Keboola\Test\Backend\Workspaces\LegacyWorkspacesSnowflakeTest::testLoadIncrementalNullable
15:13:14-UTC - paratest-backend-snowflake-abs-part-1  -	Keboola\StorageApi\ClientException: Table in.c-API-tests-984cefc1fb61bb18c6cf682cb7e826e5532278af.languages load error: Likely datatype conversion: preg_match() expects parameter 2 to be string, int given
15:13:14-UTC - paratest-backend-snowflake-abs-part-1  -	
15:13:14-UTC - paratest-backend-snowflake-abs-part-1  -	/code/src/Keboola/StorageApi/Client.php:2563
15:13:14-UTC - paratest-backend-snowflake-abs-part-1  -	/code/src/Keboola/StorageApi/Client.php:2552
15:13:14-UTC - paratest-backend-snowflake-abs-part-1  -	/code/src/Keboola/StorageApi/Client.php:2520
15:13:14-UTC - paratest-backend-snowflake-abs-part-1  -	/code/src/Keboola/StorageApi/Client.php:2407
15:13:14-UTC - paratest-backend-snowflake-abs-part-1  -	/code/src/Keboola/StorageApi/Workspaces.php:88
15:13:14-UTC - paratest-backend-snowflake-abs-part-1  -	/code/tests/Backend/Workspaces/LegacyWorkspacesSnowflakeTest.php:204
15:13:14-UTC - paratest-backend-snowflake-abs-part-1  -	/code/vendor/keboola/phpunit-retry-annotations/src/RetryTrait.php:32
15:13:14-UTC - paratest-backend-snowflake-abs-part-1  -	
15:13:14-UTC - paratest-backend-snowflake-abs-part-1  -	3) Keboola\Test\Backend\Workspaces\WorkspacesSnowflakeTest::testLoadIncrementalNullable
15:13:14-UTC - paratest-backend-snowflake-abs-part-1  -	Keboola\StorageApi\ClientException: Table in.c-API-tests-f1d3b5de9c806f917eabf5440a549950612e040d.languages load error: Likely datatype conversion: preg_match() expects parameter 2 to be string, int given
15:13:14-UTC - paratest-backend-snowflake-abs-part-1  -	
15:13:14-UTC - paratest-backend-snowflake-abs-part-1  -	/code/src/Keboola/StorageApi/Client.php:2563
15:13:14-UTC - paratest-backend-snowflake-abs-part-1  -	/code/src/Keboola/StorageApi/Client.php:2552
15:13:14-UTC - paratest-backend-snowflake-abs-part-1  -	/code/src/Keboola/StorageApi/Client.php:2520
15:13:14-UTC - paratest-backend-snowflake-abs-part-1  -	/code/src/Keboola/StorageApi/Client.php:2407
15:13:14-UTC - paratest-backend-snowflake-abs-part-1  -	/code/src/Keboola/StorageApi/Workspaces.php:88
15:13:14-UTC - paratest-backend-snowflake-abs-part-1  -	/code/tests/Backend/Workspaces/WorkspacesSnowflakeTest.php:466
15:13:14-UTC - paratest-backend-snowflake-abs-part-1  -	/code/vendor/keboola/phpunit-retry-annotations/src/RetryTrait.php:32
15:13:14-UTC - paratest-backend-snowflake-abs-part-1  -	
15:13:14-UTC - paratest-backend-snowflake-abs-part-1  -	4) Keboola\Test\Backend\Workspaces\WorkspacesSnowflakeTest::testLoadIncrementalNotNullable
15:13:14-UTC - paratest-backend-snowflake-abs-part-1  -	Keboola\StorageApi\ClientException: Table in.c-API-tests-d90ea190ca91901bbee75c5a8a69c73dbb2c49e9.languages load error: Likely datatype conversion: preg_match() expects parameter 2 to be string, int given
15:13:14-UTC - paratest-backend-snowflake-abs-part-1  -	
15:13:14-UTC - paratest-backend-snowflake-abs-part-1  -	/code/src/Keboola/StorageApi/Client.php:2563
15:13:14-UTC - paratest-backend-snowflake-abs-part-1  -	/code/src/Keboola/StorageApi/Client.php:2552
15:13:14-UTC - paratest-backend-snowflake-abs-part-1  -	/code/src/Keboola/StorageApi/Client.php:2520
15:13:14-UTC - paratest-backend-snowflake-abs-part-1  -	/code/src/Keboola/StorageApi/Client.php:2407
15:13:14-UTC - paratest-backend-snowflake-abs-part-1  -	/code/src/Keboola/StorageApi/Workspaces.php:88
15:13:14-UTC - paratest-backend-snowflake-abs-part-1  -	/code/tests/Backend/Workspaces/WorkspacesSnowflakeTest.php:562
15:13:14-UTC - paratest-backend-snowflake-abs-part-1  -	/code/vendor/keboola/phpunit-retry-annotations/src/RetryTrait.php:32
```

**Comment by martinjunger** [comment]
> [Tag](https://dev.azure.com/keboola-dev/Connection%20build/_build/results?buildId=25873&view=logs&j=ae13dfb3-00c3-5552-9b00-63e26024fcba&t=48e25ff6-be86-558e-50b2-b8f91a4d5f0f&l=1770) spadnul jen na:
> 
> ```
> 15:13:14-UTC - paratest-backend-snowflake-abs-part-1  -	1) Keboola\Test\Backend\Workspaces\LegacyWorkspacesSnowflakeTest::testLoadIncrementalNotNullable
> 15:13:14-UTC - paratest-backend-snowflake-abs-part-1  -	Keboola\StorageApi\ClientException: Table in.c-API-tests-1eb2aae730941c1153220150a9cd2609f3bef5b7.languages load error: Likely datatype conversion: preg_match() expects parameter 2 to be string, int given
> ...
> ```

Fixnuto v https://github.com/keboola/connection/pull/3991

**Comment by martinjunger** [comment]
🏷️ Tag 3: https://dev.azure.com/keboola-dev/Connection%20build/_build/results?buildId=28569&view=results

**Pull request merged**